config/config.go

```diff
@@ -466,6 +481,10 @@ func (cfg *Config) NewNode() (host.Host, error) {
 		rcmgr.MustRegisterWith(cfg.PrometheusRegisterer)
 	}
 
+	if len(cfg.PSK) > 0 && cfg.ShareTCPListener {
+		return nil, errors.New("cannot use shared TCP and WebSocket listener with PSK")
+	}
```

Comment:
LGTM, although technically the shared listener feature isn't coupled to having TCP + WS since you can currently enable it with just TCP (albeit at no real benefit) and down the road it might be configurable with HTTP as well.